src/CPUSolver.cpp

```diff
@@ -745,10 +745,10 @@ void CPUSolver::tallyScalarFlux(segment* curr_segment, int azim_index,
 
   /* Compute change in angular flux along segment in this FSR */
   for (int e=0; e < _num_groups; e++) {
-    for (int p=0; p < _num_polar; p++) {
+    for (int p=0; p < _num_polar_2; p++) {
       exponential = _exp_evaluator->computeExponential(sigma_t[e] * length, p);
       delta_psi = (track_flux(p,e)-_reduced_sources(fsr_id,e)) * exponential;
-      fsr_flux[e] += delta_psi * _polar_weights(azim_index,p);
+      fsr_flux[e] += delta_psi * _quadrature->getWeightInline(azim_index, p);
```

Comment:
I like how clean this is and that it avoids making a weights pointer attribute in the Solver class!